### PR TITLE
add soft-eviction hacks

### PR DESCRIPTION
**Pull request by BenTheElder**
this adds an experimental script to enable soft eviction thresholds on the prow bulids cluster
https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/#soft-eviction-thresholds

xref: https://github.com/kubernetes/test-infra/issues/5700#issuecomment-348387275

/area hacks
/shrug

### CODE REVIEW

**Comment by BenTheElder** [comment]
FYI @dashpole @dchen1107 @mtaufen

File: experiment/prow_builds_enable_soft_eviction.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            nodes.append(node)
+    return nodes
+
+
+def run_on_node(node_name, command):
+    print("node: %s running: %s" % (node_name, command))
+    subprocess.call(['gcloud', 'compute', 'ssh', '--project='+PROJECT, '--zone='+ZONE, '--command='+command, node_name])
+
+def main():
+    if sys.argv[-1] != "--yes-i-accept-that-this-is-very-risky":
```
Comment (mtaufen) [inline experiment/prow_builds_enable_soft_eviction.py]: +1 thanks for including a very clear warning

**Comment by BenTheElder** [comment]
Also noting that this is somewhat safer because our node pools all have [GKE node auto repair](https://cloud.google.com/kubernetes-engine/docs/concepts/node-auto-repair) enabled. So in the worst case the nodes will get drained and replace with fresh default-configured nodes.

We can try this for a bit and see how it goes. Realistically our nodes don't go down on GKE normally so we can just run this after we modify the nodes. This also isn't a hard dependency, just a nice way to reduce the chance a job is evicted due to transient disk pressure.

Also FYI @porridge @shyamjvs after a bit more testing I plan to apply this to the build cluster which should further remove ways your jobs can be evicted.

File: experiment/prow_builds_enable_soft_eviction.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        print("This command is very risky and unsupported (!)")
+        print("Do not run this unless you know what you are doing and accept the consequences (!)")
+        sys.exit(-1)
+
+    # auth to the cluster
+    print('getting cluster auth...')
+    os.system(AUTH_TO_CLUSTER_COMMAND)
+    print('')
```
Comment (krzyzacy) [inline experiment/prow_builds_enable_soft_eviction.py]: (what'd the empty print do?)

**Comment by krzyzacy** [comment]
/lgtm
/hold
as long as it works :-)

File: experiment/prow_builds_enable_soft_eviction.py (hunk 2 above)
Comment (BenTheElder) [reply experiment/prow_builds_enable_soft_eviction.py]: just space out the output a bit :-) (prints a blank line)

**Comment by BenTheElder** [comment]
/hold cancel
running this now :fire:
FYI @cjwagner

**Comment by BenTheElder** [comment]
per @mtaufen checked against kubectl proxy at `api/v1/proxy/nodes/gke-prow-pool-n1-highmem-8-81ce4395-zl18/configz` (one of the build nodes) which contains the snippet:
`"evictionSoft":"nodefs.available\u003c30","evictionSoftGracePeriod":"nodefs.available=2h",`
To further verify this.

**Pull request merged**